pkg/network/server_test.go

```diff
@@ -133,7 +133,7 @@ func TestServerRegisterPeer(t *testing.T) {
 		ps[i].version = &payload.Version{Nonce: uint32(i), UserAgent: []byte("fake")}
 	}
 
-	startWithCleanup(t, s)
+	go s.Start()
```

Comment:
OK, if server doesn't wait for all goroutines to finish, then this commit is useless, it won't fix the issue.

Reply:
With require.Eventually at the end of the tests it will fix races in tests, but agree the waiting of goroutines should be in shutdown (working on it). t.Cleanup does not work well, tried it here https://github.com/nspcc-dev/neo-go/commit/be9ce6ceb352008daf576297f8dbf1c5dbce15f9 and sometimes before, not sure that t.Cleanup is useful

Reply:
Let's keep `t.Cleanup(s.Shutdown())` for now (the unchanged version of `startWithCleanup`), I think it will work properly with proper shutdown awaiting in the network server.